### PR TITLE
TM-4649 & TM-4562: Add Override TOD and Functional Bureau to Publishable Position card 

### PR DESCRIPTION
**Pull request by pveland621**
Pulled the TODs and functional bureaus from the filters already in redux (assuming they will always be loaded before this page so i didn't add a fetch call to this component)

ACs:
1. Verify new override TOD dropdown appears on editable publishable position cards
2. Verify new functional bureaus section appears on editable publishable position cards

[TOD Ticket](https://metaphase.atlassian.net/browse/TM-4649)
[Functional Bureaus Ticket](https://metaphase.atlassian.net/browse/TM-4652)

### CODE REVIEW

File: src/sass/_positionContent.scss [hunk 1]
```diff
@@ -245,6 +245,7 @@
     display: flex;
     align-items: baseline;
     flex-direction: column;
+    padding-right: 20px;
```
[the remark this answers is not in the document]
Comment (pveland621) [reply src/sass/_positionContent.scss:248]: omg thank u thank u

File: src/sass/_positionContent.scss [hunk 2]
```diff
@@ -313,7 +311,17 @@
   .spaced-row {
     display: flex;
     justify-content: space-between;
-    align-items: flex-end;
+    align-items: center;
+
+    .dropdown-container {
+      display: flex;
+      justify-content: flex-start;
+      align-items: flex-end;
```
Comment (elizabeth-jimenez) [inline src/sass/_positionContent.scss:319]: Wondering if we need these two styles

```
justify-content: flex-start;
align-items: flex-end;
```

File: src/Components/PublishablePositionCard/PublishablePositionCard.test.jsx [hunk 3]
```diff
@@ -1,12 +1,22 @@
...
+const middlewares = [thunk];
+const mockStore = configureStore(middlewares);
+
 describe('PublishablePositionCard', () => {
   const result = resultsObject.results[0];
 
   it('is defined', () => {
-    const wrapper = shallow(<PublishablePositionCard data={result} />);
+    const wrapper = shallow(
+      <Provider store={mockStore({})}>
```
Comment (elizabeth-jimenez) [inline src/Components/PublishablePositionCard/PublishablePositionCard.test.jsx:16]: 🙌

File: src/Components/PublishablePositionCard/PublishablePositionCard.jsx [hunk 4]
```diff
@@ -172,6 +197,27 @@ const PublishablePositionCard = ({ data, cycles, onEditModeSearch }) => {
         valueKey="id"
         labelKey="name"
       />
+      <div className="functional-bureau-section">
```
Comment (elizabeth-jimenez) [inline src/Components/PublishablePositionCard/PublishablePositionCard.jsx:200]: I think you'll want to add to the Static portion too

File: src/Components/PublishablePositionCard/PublishablePositionCard.jsx [hunk 5]
```diff
@@ -113,19 +122,35 @@ const PublishablePositionCard = ({ data, cycles, onEditModeSearch }) => {
...
+            >
+              {statusOptions.map(s => (
+                <option value={s.code}>
+                  {s.name}
+                </option>
+              ))}
+            </select>
+          </div>
+          <div className="position-form--input">
+            <label htmlFor="publishable-pos-tod-override">Override TOD</label>
```
Comment (elizabeth-jimenez) [inline src/Components/PublishablePositionCard/PublishablePositionCard.jsx:141]: Just a styling question - wondering if we should spell out TOD like we did in the other field

File: src/Components/PublishablePositionCard/PublishablePositionCard.jsx [hunk 6]
```diff
@@ -89,6 +90,14 @@ const PublishablePositionCard = ({ data, cycles, onEditModeSearch }) => {
   const [exclude, setExclude] = useState(true);
```
Comment (jlam-mpc) [inline src/Components/PublishablePositionCard/PublishablePositionCard.jsx:90]: Confused by the AC in the ticket, are we supposed to show Functional Bureau in the static card as well? (Referring to Line 52 not this line)

**Comment by elizabeth-jimenez** [comment]
Approved Pending Feedback ⏱️:
AC: Add “Functional Bureau” field to Expanded card view of the Publishable Position cards

File: src/Components/CyclePositionCard/CyclePositionCard.jsx [hunk 7]
```diff
@@ -139,20 +139,18 @@ const CyclePositionCard = ({ data, cycle, onEditModeSearch }) => {
...
-                onChange={(e) => setIncumbent(e?.target.value)}
-              >
-                {fakeIncumbents.map(s => (
-                  <option value={s.code}>
-                    {s.name}
-                  </option>
-                ))}
-              </select>
-            </div>
+            <label htmlFor="cycle-position-incumbent">Incumbent</label>
```
Comment (jlam-mpc) [inline src/Components/CyclePositionCard/CyclePositionCard.jsx:142]: Nice cleanup

File: src/Components/PublishablePositionCard/PublishablePositionCard.jsx (hunk 4 above)
Comment (jlam-mpc) [reply src/Components/PublishablePositionCard/PublishablePositionCard.jsx:200]: ^^ Remember to put it in `bodySecondary` of `sections` not `bodyPrimary`!

File: src/sass/_positionContent.scss (hunk 2 above)
Comment (jlam-mpc) [reply src/sass/_positionContent.scss:319]: I don't think so either

File: src/sass/_positionContent.scss [hunk 8]
```diff
@@ -313,7 +311,17 @@
...
+    }
+  }
+
+  .functional-bureau-section {
+    margin-top: 20px;
```
Comment (jlam-mpc) [inline src/sass/_positionContent.scss:324]: If it doesn't need to be styled any more than this, wondering if we should just throw in a `.mt-20` in `_common.scss` and use that (or use the existing `.pt-20`).

**Review by jlam-mpc — COMMENTED** [review]
Functions great and there was good catches and nice cleanup too! Just missing the static Functional Bureau in the expanded read only card and a minor suggestion on simplifying styling/classnames.

File: src/sass/_positionContent.scss (hunk 2 above)
Comment (pveland621) [reply src/sass/_positionContent.scss:319]: get em outta here

File: src/sass/_positionContent.scss (hunk 8 above)
Comment (pveland621) [reply src/sass/_positionContent.scss:322]: noice suggestion - i'm all for reducing styling bloat. changed!

**Comment by pveland621** [comment]
pretty sure i addressed the static feedback, let me know if i still missed something~

File: src/Components/PublishablePositionCard/PublishablePositionCard.jsx [hunk 9]
```diff
@@ -109,23 +119,40 @@ const PublishablePositionCard = ({ data, cycles, onEditModeSearch }) => {
       'Tour of Duty': getResult(pos, 'post.tour_of_duty') || NO_TOUR_OF_DUTY,
       'Pay Plan': '---',
       'Assignee': '---',
+      'Functional Bureau': 'None Listed',
```
Comment (elizabeth-jimenez) [inline src/Components/PublishablePositionCard/PublishablePositionCard.jsx:122]:
```suggestion
```

**Review by elizabeth-jimenez — APPROVED** [review]

**Pull request merged**